cloudinit/config/schema.py

```diff
@@ -135,28 +339,55 @@ def get_jsonschema_validator():
     # http://json-schema.org/understanding-json-schema/reference/object.html#pattern-properties
     strict_metaschema["properties"]["label"] = {"type": "string"}
 
+    validator_kwargs = {}
     if hasattr(Draft4Validator, "TYPE_CHECKER"):  # jsonschema 3.0+
         type_checker = Draft4Validator.TYPE_CHECKER.redefine(
             "string", is_schema_byte_string
         )
-        cloudinitValidator = create(
-            meta_schema=strict_metaschema,
-            validators=Draft4Validator.VALIDATORS,
-            version="draft4",
-            type_checker=type_checker,
-        )
+        validator_kwargs = {
+            "type_checker": type_checker,
+        }
     else:  # jsonschema 2.6 workaround
         types = Draft4Validator.DEFAULT_TYPES  # pylint: disable=E1101
         # Allow bytes as well as string (and disable a spurious unsupported
         # assignment-operation pylint warning which appears because this
         # code path isn't written against the latest jsonschema).
         types["string"] = (str, bytes)  # pylint: disable=E1137
-        cloudinitValidator = create(  # pylint: disable=E1123
-            meta_schema=strict_metaschema,
-            validators=Draft4Validator.VALIDATORS,
-            version="draft4",
-            default_types=types,
+        validator_kwargs = {"default_types": types}
+
+    # Add deprecation handling
+    SchemaDeprecationError = _DeprecationErrorCreator().error_type
+    validators = dict(Draft4Validator.VALIDATORS)
+    validators[DEPRECATED_KEY] = partial(
+        _validator_deprecated, error_type=SchemaDeprecationError
+    )
+    validators["oneOf"] = partial(_oneOf, error_type=SchemaDeprecationError)
+    validators["anyOf"] = partial(_anyOf, error_type=SchemaDeprecationError)
+    validators["allOf"] = partial(_allOf, error_type=SchemaDeprecationError)
+
+    cloudinitValidator = create(
+        meta_schema=strict_metaschema,
+        validators=validators,
+        version="draft4",
+        **validator_kwargs,
+    )
+
+    # Add deprecation handling
+    def is_valid(self, instance, _schema=None, **__):
```

Comment:
Does this mean that our schema will no longer be valid against the built-in validators?

Reply:
No, it does not. We needed because we add extra meaning to the `deprecated` key, which in `draft4` is not defined and in newer drafts is treated as an [annotation](https://json-schema.org/understanding-json-schema/reference/generic.html?highlight=deprecated#annotations).

Other validators will ignore `deprecated` and `deprecated.time`.